sdk/ai/azure-ai-projects/tests/test_datasets_async.py

```diff
@@ -0,0 +1,206 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+import re
+
+# import pytest
+from azure.ai.projects.aio import AIProjectClient
+from azure.ai.projects.models import DatasetVersion, DatasetType
+from test_base import TestBase, servicePreparer
+from devtools_testutils.aio import recorded_by_proxy_async
+from devtools_testutils import is_live_and_not_recording
+from azure.core.exceptions import HttpResponseError
+
+
+# Construct the paths to the data folder and data file used in this test
+script_dir = os.path.dirname(os.path.abspath(__file__))
+data_folder = os.environ.get("DATA_FOLDER", os.path.join(script_dir, "test_data/datasets"))
+data_file1 = os.path.join(data_folder, "data_file1.txt")
+data_file2 = os.path.join(data_folder, "data_file2.txt")
+
+
+class TestDatasetsAsync(TestBase):
+
+    # To run this test, use the following command in the \sdk\ai\azure-ai-projects folder:
+    # cls & pytest tests\test_datasets_async.py::TestDatasetsAsync::test_datasets_upload_file_async -s
+    @servicePreparer()
+    @recorded_by_proxy_async
+    # TODO: Why doesn't this work? @pytest.mark.skipif(not is_live_and_not_recording(), reason="Skipped because this test involves network calls from another client (azure.storage.blob) that is not recorded.")
```

Comment:
Please try to put this decorator between `@servicePreparer()` and `@recorded_by_proxy_async`.

Reply:
Thank you @nick863 , this worked!! I switched to using the decorator.